dt-demo/dimensions/src/test/java/com/datatorrent/demos/dimensions/telecom/CustomerServiceHbaseOutputOperatorTester.java

```diff
@@ -39,31 +61,259 @@ public void test() throws Exception
+            Bytes.toBoolean(nameValueMap.get("satisfied")));
+        fetchedCount++;
+        
+        Assert.assertTrue("Don't have data: " + cs, cs.equals(imsiToCsMap.get(cs.imsi)));
+      }
+      while(result != null);
+    }while(hasResult);
+    
+    Assert.assertTrue("The number of records read from HBase (" + fetchedCount + ") is not same as expected (" + imsiToCsMap.size() + ").", fetchedCount == imsiToCsMap.size());
+  }
+  
+  public static void waitMills(int millSeconds)
+  {
+    if(millSeconds <= 0)
+      return;
+    try
+    {
+      Thread.sleep(millSeconds);
+    }
+    catch(Exception e)
+    {
+      //ignore
+    }
+  }
+  
+//  
+//  
+//  public static class SpeCustomerServiceHbaseOutputOperator extends CustomerServiceHbaseOutputOperator
+//  {
+//    protected long totalReceivedTuples = 0;
+//    protected long totalCommittedTuples = 0;
+//
+//    @Override
+//    public void setup(OperatorContext context)
+//    {
+//      super.setup(context);
+//      
+//      store.getTable().setAutoFlushTo(false);
+//    }
+//    
+//    protected Set<String> imsiSet = Sets.newHashSet();
+//    
+//    @Override
+//    public void processTuple(CustomerService tuple)
+//    {
+//      if(!imsiSet.add(tuple.imsi))
+//        return;
+//      
+//      totalReceivedTuples++;
+//
+//      HTable table = store.getTable();
+//      Put put = operationPut(tuple);
+//      try {
+//        table.put(put);
+//        ++unCommittedSize;
+////        if (++unCommittedSize >= 1000) {
+////          table.flushCommits();
+////          totalCommittedTuples += unCommittedSize;
+////          unCommittedSize = 0;
+////        }
+//      } catch (RetriesExhaustedWithDetailsException e) {
+//        logger.error("Could not output tuple", e);
+//        DTThrowable.rethrow(e);
+//      } catch (InterruptedIOException e) {
+//        logger.error("Could not output tuple", e);
+//        DTThrowable.rethrow(e);
+//      }
+//
+//    }
+//
+//    @Override
+//    public void endWindow()
+//    {
+//      try {
+//        if (unCommittedSize > 0) {
+//          store.getTable().flushCommits();
+//          
+//          totalCommittedTuples += unCommittedSize;
+//          unCommittedSize = 0;
+//        }
+//      } catch (RetriesExhaustedWithDetailsException e) {
+//        logger.error("Could not output tuple", e);
+//        DTThrowable.rethrow(e);
+//      } catch (InterruptedIOException e) {
+//        logger.error("Could not output tuple", e);
+//        DTThrowable.rethrow(e);
+//      }
+//
+//      logger.info("Total Received: {}, Total committed: {}.", totalReceivedTuples, totalCommittedTuples);
+//    }
+//
+//  }
+  
+  public static class SpecificCustomerServiceGenerateOperator extends CustomerServiceGenerateOperator
+  {
+    protected String dataId = "SpecificCustomerServiceGenerateOperator";
+    protected long capacity = 100;
+    protected AtomicBoolean terminateFlag;
```

Comment:
Do you mean "SpecificCustomerServiceGenerateOperator". Yes if it is. This operator can set a flag when required tuples have been sent. And the control code can use this flag to decide when to shutdown the DAG